gen/solid.hoon

```diff
@@ -11,26 +11,44 @@
   !:
 :-  %say
 |=  $:  {now/@da eny/@uvJ bec/beak}
-        {$~ $~}
+        {arg/$@($~ {top/path $~}) dub/_|}
     ==
 :-  %noun
-=+  top=`path`/(scot %p p.bec)/[q.bec]/(scot %da now)/sys
-=+  pax=`path`(weld top `path`[%hoon ~])
-=+  arp=`path`(weld top `path`[%ovra ~])
+?~  arg  $(arg ~[top=`path`/(scot %p p.bec)/[q.bec]/(scot %da now)/sys])
+=+  pax=`path`(weld top.arg `path`[%hoon ~])
+=+  arp=`path`(weld top.arg `path`[%ovra ~])
 ~&  %solid-start
 =+  txt=.^(@t %cx (weld pax `path`[%hoon ~]))
 =+  rax=.^(@t %cx (weld arp `path`[%hoon ~]))
 =+  ^=  ken
-    =-  ?:(?=($& -.res) p.res ((slog p.res) ~))
+    =-  ?:(?=($& -.res) p.res (mean (flop p.res)))
     ^=  res  %-  mule  |.
     ~&  %solid-loaded
     =+  gen=(rain pax txt)
     ~&  %solid-parsed
     =+  one=(~(mint ut %noun) %noun gen)
     ~&  %solid-compiled
-    =+  two=(~(mint ut p.one) %noun (rain arp rax))
-    ~&  %solid-arvo
-    [7 q.one q.two]
+    ?.  dub
+      =+  two=(~(mint ut p.one) %noun (rain arp rax))
+      ~&  %solid-arvo
+      [7 q.one q.two]
+    =/  tri
+      '''
+      :: XX moveme to, uh arvo probably, this depends on too many names
+      |=  [pax=path txt=@t arp=path rax=@t]
+      =+  gen=(rain pax txt)
+      ~&  %solid-double-parsed
+      =+  one=(~(mint ut %noun) %noun gen)
+      ~&  %solid-double-compiled
+      =+  two=(~(mint ut p.one) %noun (rain arp rax))
+      ~&  %solid-arvo
+      [7 q.one q.two]
+      '''
+    =+  all=.*(0 q.one)
+    .*  all
+    :+  7  =<(+ .*(all [9 2 0+2 1+[p.one tri] 0+7]))
```

Comment:
Calls the hoon compiler gate, replacing the sample with the type of hoon.hoon (`p.one`) and the cord containing the code to run the second `|solid`. This produces a pair of type and nock, so we grab just the nock using `+`.